platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/api/ResolutionResultApiIntegrationTest.groovy

```diff
@@ -1076,4 +1076,39 @@ testRuntimeClasspath
         expect:
         succeeds("traverse")
     }
+
+    def "attributes on resolved variants can be requested using strongly-typed values"() {
+        settingsFile << """
+            include("producer")
+        """
+
+        file("producer/build.gradle") << """
+            plugins {
+                id("java-library")
+            }
+        """
+
+        buildFile << """
+            plugins {
+                id("java-library")
+            }
+
+            dependencies {
+                implementation project(":producer")
+            }
+
+            task resolve {
+                def root = configurations.runtimeClasspath.incoming.resolutionResult.rootVariant
```

Comment:
I don't think this test is verifying the changes we made here. Resolution result variant attributes should already be desugared. 

We want to verify the attributes of something like:
```
    configuration.getIncoming().getArtifacts().getArtifacts().getFirst().getVariant()

```

I would be curious to run this test without any of the production code changes. Does it fail then?

Reply:
Added and expanded tests to get at various sets of resolved attributes in various ways and confirm both strongly and strongly typed attributes can be retrieved.